### PR TITLE
feat:add kitex template

### PR DESCRIPTION
**Pull request by ViolaPioggia**
#### What type of PR is this?
feat
#### Check the PR title.
feat:add kitex template
#### (Optional) Translate the PR title into Chinese.
增加 template 模版的示例

### CODE REVIEW

File: kitex/template/go.mod [hunk 1]
```diff
@@ -0,0 +1,5 @@
+module github.com/kitex/hello
+
+go 1.21
+
+replace github.com/apache/thrift => github.com/apache/thrift v0.13.0
```
Comment (felix021) [inline kitex/template/go.mod:5]: no dependency?

File: kitex/template/readme_zn.md [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+| main.go     | 启动文件        |
+| handler.go  | 用于请求处理返回响应。 |
+| kitex_gen   | kitex 生成的代码 |
+| biz/service | 实际的业务逻辑。    |
+| biz/dal     | 存储层操作逻辑     |
+
+## 如何运行
+
+````外壳
+sh 构建.sh
```
Comment (felix021) [inline kitex/template/readme_zn.md:24]: file names shouldn't be translated.

File: kitex/template/script/bootstrap.sh [hunk 3]
```diff
@@ -0,0 +1,4 @@
+#! /usr/bin/env bash
+CURDIR=$(cd $(dirname $0); pwd)
+echo "$CURDIR/bin/"
+exec "$CURDIR/bin/"
```
Comment (felix021) [inline kitex/template/script/bootstrap.sh:4]: doesn't seem capable of executing the binary.

File: kitex/template/template/bootstrap_sh_tpl.yaml [hunk 4]
```diff
@@ -0,0 +1,8 @@
+path: script/bootstrap.sh
+update_behavior:
+  type: skip
+body: |-
+  #! /usr/bin/env bash
+  CURDIR=$(cd $(dirname $0); pwd)
+  echo "$CURDIR/bin/{{.RealServiceName}}"
+  exec "$CURDIR/bin/{{.RealServiceName}}"
```
Comment (felix021) [inline kitex/template/template/bootstrap_sh_tpl.yaml:8]: So where is the RealServiceName assigned?

Comment (felix021) [reply kitex/template/template/bootstrap_sh_tpl.yaml:8]: Try to make it simpler to avoid confusion

File: kitex/template/conf/dev/conf.yaml [hunk 5]
```diff
@@ -0,0 +1,23 @@
+kitex:
+  service: ""
```
Comment (felix021) [inline kitex/template/conf/dev/conf.yaml:2]: service name is also missing

File: kitex/template/.gitignore
Comment (AsterDY) [inline kitex/template/.gitignore:1]: 目录结构把template放到生成的demo里面有点奇怪，能不能把生成代码再包一层文件夹和template并列

File: README_CN.md [hunk 6]
```diff
@@ -59,6 +59,7 @@
 
 ## Kitex 生成代码
 - [protobuf](kitex/protobuf) 使用 kitex 与 protobuf 生成服务端代码的示例
+- [template](kitex/template) 使用 kitex 自定义模版生成服务端代码的示例
```
Comment (AsterDY) [inline README_CN.md:62]: 然后kitex/template再加一个reame或者script说明一下怎么使用这个tmpl生成吧

Comment (ViolaPioggia) [reply README_CN.md:62]: template 目录下有 makefile 文件

File: kitex/template/.gitignore
Comment (ViolaPioggia) [reply kitex/template/.gitignore:1]: hertz-example 里就是这样安排的，是根据 template 模版直接生成的代码

**Review by li-jin-gou — CHANGES_REQUESTED** [review]
CI 没过

**Review by li-jin-gou — APPROVED** [review]

**Pull request merged**